tinygrad/codegen/linearizer.py

```diff
@@ -249,10 +250,11 @@ def global_load(self, i:int, idxs:Sequence[VariableOrNum], const=None) -> List[T
         idx, valid = self.sts[i].expr_idxs(_idx)
         localtype = dtypes.float32
       key = f"{localtype}{idx.render()}{valid.render()}"
+      this_const, valid, key = (invalid_value, cast(Variable, Variable.num(1)), f"{localtype}{valid.render()}") if valid.max == 0 else (const, valid, key)
```

Comment:
Why does this need a cast? Why does the key render valid? It's just 0!

Reply:
The cast is because NumNode(k) is not a Variable... :( Not sure why MemOp/ConstOp has Variables instead of Nodes, is it ok to change that?

Reply:
I changed the valid render, but this is just temporary -- when this merges with #1482 I will just treat invalid as RawConst 0 valid 1.